storage_delegation/main.tf

```diff
@@ -6,16 +12,48 @@ locals {
     "eu-de"    = "eu-de.dataplatform.cloud.ibm.com",
     "jp-tok"   = "jp-tok.dataplatform.cloud.ibm.com"
   }
-  dataplatform_ui = local.dataplatform_ui_mapping[local.location]
+  dataplatform_ui             = local.dataplatform_ui_mapping[local.location]
+  kms_service                 = module.crn_parser.service_name
+  kms_account_id              = module.crn_parser.account_id
+  kms_key_id                  = module.crn_parser.resource
+  target_resource_instance_id = split(":", var.cos_kms_crn)[7]
```

Comment:
the crn parser has support to parse the resource instance. (its the `service_instance` output)

Reply:
This has been implemented